lib/features/home_page/home_page.dart

```diff
@@ -65,14 +75,48 @@ class HomePage extends StatelessWidget {
           mainAxisSpacing: 10,
         ),
         itemCount: pokemon?.length,
-        itemBuilder: (context, index) => PokemonTile(pokemon: pokemon?[index]),
+        itemBuilder: (context, index) => PokemonTile(
+          pokemon: pokemon?[index],
+          key: ValueKey(pokemon?[index].name),
+        ),
       ),
-      floatingActionButton: FloatingActionButton(
-        onPressed: () {
-          // TODO:
-        },
-        backgroundColor: red,
-        child: const Icon(Icons.filter_list),
+      floatingActionButton: Row(
+        mainAxisAlignment: MainAxisAlignment.end,
+        children: [
+          Visibility(
```

Comment:
You can just use the checking `(isVisible)` to show the FAB. 

`if (isVisible) FloatingActionButton`